pkg/bootstrapper/bootstrapper.go

```diff
@@ -271,50 +276,133 @@ func (wmcb *winNodeBootstrapper) translateFile(ignitionSource string, fileTransl
+				continue
+			}
+			keyValSplit := strings.Split(split[1], "=")
+			if len(keyValSplit) != 2 {
+				continue
+			}
+			envVars[keyValSplit[0]] = keyValSplit[1]
+		}
+	}
+	return envVars
+}
+
+// getEnvVarName returns the name of an environment variable located in the given string. The variable must be specified
+// by the format ${var_name}. There can only be one environment variable in the given string.
+func getEnvVarName(full string) (string, error) {
+	envVarSplit := strings.Split(full, "${")
+	if len(envVarSplit) != 2 {
+		return "", fmt.Errorf("could not find exactly one opening sequence")
+	}
+	trimSplit := strings.Split(envVarSplit[1], "}")
+	if len(trimSplit) != 2 {
+		return "", fmt.Errorf("could not find exactly one closing brace")
+	}
+	return trimSplit[0], nil
+}
+
+// resolveKubeletArg cleans the input value returns the true value of it, either unchanged or resolved via the provided
+// map
+func resolveKubeletArg(value string, envVars map[string]string) (string, error) {
+	// Line can contain an escaped newline which leaves the substring with a trailing "\"
+	foundValue := strings.TrimSuffix(value, "\\")
+	trimmedValue := strings.TrimSpace(foundValue)
+	if varName, err := getEnvVarName(trimmedValue); err == nil {
+		// arg value is an environment variable so use the environment variable value
+		varValue, ok := envVars[varName]
+		if !ok {
+			return "", fmt.Errorf("arg is set to a env variable that was not set")
+		}
+		trimmedValue = varValue
+	}
+	return trimmedValue, nil
+}
+
+// parseSystemdKubeletUnit gathers the kubelet args we need from the kubelet systemd unit file and returns a
+// map of file translations to be added to the existing file translation map
+func (wmcb *winNodeBootstrapper) parseSystemdKubeletUnit(contents string) (map[string]fileTranslation, error) {
+	var err error
+	files := make(map[string]fileTranslation)
+	envVars := extractEnvVarsFromSystemdUnit(contents)
+
+	// For each of these regex submatches we are grabbing a string that can either be a regular string:
+	// --v=3
+	// or an environment variable:
+	// --v="${KUBELET_LOG_LEVEL}"
+	results := cloudProviderRegex.FindStringSubmatch(contents)
+	if len(results) == 2 {
+		wmcb.kubeletArgs[cloudProviderOption], err = resolveKubeletArg(results[1], envVars)
+		if err != nil {
+			return files, fmt.Errorf("could not resolve kubelet arg %s: %s", cloudProviderOption, err)
+		}
+	}
+	results = verbosityRegex.FindStringSubmatch(contents)
+	if len(results) == 2 {
+		wmcb.kubeletArgs[verbosityOption], err = resolveKubeletArg(results[1], envVars)
+		if err != nil {
+			return files, fmt.Errorf("could not resolve kubelet arg %s: %s", verbosityOption, err)
+		}
+	}
+	// Check for the presence of "--cloud-config" option and if it is present append the value to
+	// filesToTranslate. This option is only present for Azure and hence we cannot assume it as a file that
+	// requires translation across clouds.
+	results = cloudConfigRegex.FindStringSubmatch(contents)
```

Comment:
I see a pattern here, can we just create a function around this instead of duplicated code blocks?

Reply:
I don’t see you addressing @ravisantoshgudimetla’s request.

Reply:
We talked in person, I should have updated the comment. I asked @ravisantoshgudimetla how he could see a function extracted here. He was unable to tell me at the time. I'll follow up